src/formatDate.js

```diff
@@ -8,7 +8,32 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat[3];
+  const dateValues = date.split(separator);
+  const dateParts = {};
+
+  for (let i = 0; i < dateValues.length; i++) {
+    dateParts[fromFormat[i]] = dateValues[i];
+  }
+
+  if (dateParts['YYYY']) {
+    dateParts['YY'] = dateParts['YYYY'].slice(-2);
+  } else if (dateParts['YY'] < 30) {
+    dateParts['YYYY'] = `20${dateParts['YY']}`;
+  } else if (dateParts['YY'] >= 30) {
+    dateParts['YYYY'] = `19${dateParts['YY']}`;
+  }
+
+  const newDateArr = [];
+  const newSeparator = toFormat[3];
```

Comment:
Similar to the issue in line 11, using a hardcoded index (3) to access the separator in `toFormat` assumes a fixed position. Ensure that the separator's position is correctly identified or passed as a separate parameter.